rclcpp/src/rclcpp/parameter_value.cpp

```diff
@@ -146,12 +146,30 @@ ParameterValue::ParameterValue(const int int_value)
   value_.type = rcl_interfaces::msg::ParameterType::PARAMETER_INTEGER;
 }
 
+ParameterValue::ParameterValue(const unsigned int int_value)
+{
+  if (int_value > INT32_MAX) {
+    throw std::runtime_error("Can not set a unsinged parameter to negitive value");
+  }
```

Comment:
We intentionally did not support unsigned integers being assigned into the signed integer parameters in order to avoid checks like this.